frontend/src/component/feature/FeatureToggleList/FeatureToggleFilters/AddFilterButton.tsx

```diff
@@ -29,15 +29,11 @@ const AddFilterButton = ({
     };
 
     const onClick = (label: string) => {
-        const filters = availableFilters.map((filter) =>
-            filter.label === label
-                ? {
-                      ...filter,
-                      enabled: true,
-                  }
-                : filter,
-        );
-        setAvailableFilters(filters);
+        const filterVisibility = {
```

Comment:
Simplifies state change, since we keeping just a Map, instead of complex objects.